packages/@react-aria/datepicker/src/useDatePicker.ts

```diff
@@ -147,6 +147,7 @@ export function useDatePicker<T extends DateValue>(props: AriaDatePickerProps<T>
       'aria-labelledby': `${buttonId} ${labelledBy}`,
       'aria-describedby': ariaDescribedBy,
       'aria-expanded': state.isOpen,
+      isDisabled: props.isDisabled,
```

Comment:
this should be
```suggestion
      isDisabled: props.isDisabled || props.isReadOnly,
```
Same as here https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/datepicker/src/DatePicker.tsx#L160

Once we do that here, we can delete it in the file I linked, and then readonly will work as expected in the RAC version as well